app/controllers/package_controller.rb

```diff
@@ -92,30 +94,66 @@ def category
 
 
   def screenshot
-    required_parameters :package
+    required_parameters :package, :appscreen
     package = params[:package]
+    appscreen = params[:appscreen]
     default_url = File.join( Rails.root, "app/assets/images/default-screenshots/no_screenshot_opensuse_big.png" )
-    image package, "screenshot", default_url
+    image package, "screenshot", default_url, appscreen
   end
 
   def thumbnail
-    required_parameters :package
+    required_parameters :package, :appscreen
     package = params[:package]
-    default_url = File.join( Rails.root, "app/assets/images/default-screenshots/no_screenshot_opensuse.png" )
-    image package, "thumbnail", default_url
+    appscreen = params[:appscreen]
+
+    case package
+    when /-devel$/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-devel-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-lang$/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-debug$/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-doc$/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/files.png" )
+    when /-help-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/files.png" )
+    when /-javadoc$/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/files.png" )
+    when /-debuginfo/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-debugsource/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /-kmp-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/file_settings.png" )
+    when /^rubygem-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/rubygem.png" )
+    when /^perl-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/perl.gif" )
+    when /^python-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/python.png" )
+    when /^kernel-/
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/tux.png" )
+    when /^openstack-/i
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/openstack.png" )
+    else
+      default_url = File.join( Rails.root, "app/assets/images/default-screenshots/no_screenshot_opensuse.png" )
+    end
```

Comment:
Agreed. My main problem was that I had trouble accessing `appdata` in those helper functions, but this can probably easily be resolved with some more advanced knowledge on Ruby.